.github/workflows/codeql.yml

```diff
@@ -0,0 +1,99 @@
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron: '19 9 * * 3'
+
+permissions: {}
+
+jobs:
+  analyze-host-x86_64-release:
```

Comment:
I'm a little concerned this essentially duplicates jobs we have already. I already did that for the cache seeding and regret the overhead of maintaining it. It feels like there should be a better way e.g. same workflow with different inputs or workflows calling workflows, or conditionally setting up env variables.

Reply:
From the PoV of CodeQL we just need to build the source code between the `Initialize CodeQL` and `Perform CodeQL Analysis` jobs. I lack the context of the project so can't say with certainity, but it sounds as if we had different actions for each configuration, instead of using the same actions+parameters we may reuse those 'alias' actions. 

A second option could be having a single actions script with a configuration dictionary indexed with a descriptive key. So we always use the same script with a single parameter (the configuration dictionary key), so we can use it in a workflow matrix. 

```yaml
  strategy:
    matrix: 
      include:
      - configuration: build-host-x86_64-release
      - configuration: build-riscv-m1-release
      - configruation: ...
...
jobs:

  uses: ./.github/actions/build_and_run_ock
  with:
     configuration: ${{ matrix.configuration }}
```

Both options add complexity and need refactoring the workflows so maybe it's not worth it. One thing I tried was creating a matrix that changed the `uses: ` action, but it's not allowed.

Reply:
I think we can allow this one through on the basis we will do the same thing in a different way - it sounds like your requirements are to just build it but not run it (which coincidentally is the same requirement that building the PR cache has).

I'll look into this problem.